dev/inc/RuntimeClassHelpers.h

```diff
@@ -134,24 +134,35 @@ struct ReferenceTracker : public ImplT<D, I ..., ::IReferenceTrackerExtension>,
     // return false.  If we're off the UI thread but can't get to it, then do the DeleteInstance() here (asynchronously).
     static void DeleteInstanceOnUIThread(std::unique_ptr<D>&& self) noexcept
     {
-        bool queued = false;
-        
         // See if we're on the UI thread
         if(!self->IsOnThread())
         {
+            struct LoggingState
+            {
+                void* dispatcherQueueWhenLambdaRan{ nullptr };
+                void* coreDispatcherWhenLambdaRan{ nullptr };
+                std::atomic<int> runCount;
+            };
+            auto loggingState = std::make_shared<LoggingState>();
```

Comment:
I think this could be achieved with a mutable lambda without having to create a shared pointer.